woocommerce/woocommerce-ios
# Login Onboarding: only shown if the user hasn't finished onboarading

Part of #7284 


### Description


For the login onboarding experiment, we don't want to show it again if the user has finished the onboarding either by skipping it or tapping "Next" which leads to the pre-existing prologue screen. To save the app state for whether the user has finished the onboarding, we normally dispatch a store action like `AppSettingsAction` but we can't use `StoresManager` in the logged-out state because `DeauthenticatedState` is used and store actions are no-op. As a result, UserDefaults are often used in the logged-out state. In this PR, a protocol `LoggedOutAppSettingsProtocol` was created with `LoggedOutAppSettings` implementation that takes in a `UserDefaults`. `AppCoordinator` then uses `LoggedOutAppSettingsProtocol` to determine whether to present the onboarding screen after displaying the authentication UI.


### Testing instructions


Prerequisite: if you have tested this branch before, please delete and reinstall the app to clear all user defaults.

- Launch the app
- Log out if needed --> login onboarding screen should be shown
- Tap "Next" only twice to go to the last feature page on the onboarding screen so that it is not dismissed
- Relaunch the app --> login onboarding screen should be shown again
- Tap "Next" three times or tap "Skip" to go to the prologue screen
- Relaunch the app --> login onboarding screen should not be shown
- Log in to the app
- Log out --> login onboarding screen should not be shown

---
